tests/features/bootstrap/FeatureContext.php

```diff
@@ -681,4 +681,52 @@ public static function cleanCommunityEnvironment() {
     drupal_flush_all_caches();
   }
 
+  /**
+   * Create a new revision for content of given type and title.
+   *
+   * @Given I create a new revision for :arg1 content with title :arg2
+   */
+  public function createNewRevision($type, $title) {
+    $node = $this->getNodeByTitle($type, $title);
+    $node->revision = TRUE;
+    node_save($node);
+  }
+
+  /**
+   * Revert given content to its first revision.
+   *
+   * @Given I revert the :arg1 content with title :arg2 to its first revision
+   */
+  public function revertContentToFirstRevision($type, $title) {
+    $node = $this->getNodeByTitle($type, $title);
+    $revisions = node_revision_list($node);
+    ksort($revisions);
+    $node_revision = array_shift($revisions);
+    $node_revision = node_load($node->nid, $node_revision->vid);
+    $node_revision->revision = TRUE;
+    node_save($node_revision);
+  }
+
+  /**
+   * Assert field language given field name, content type and content title.
+   *
+   * @Then I should only have :arg1 in :arg2 for :arg3 content with title :arg4
+   */
+  public function assertFieldLanguageForContentWithTitle($field_name, $language, $type, $title) {
+    $node = $this->getNodeByTitle($type, $title);
+
+    $query = db_select("field_data_{$field_name}", 'f')
```

Comment:
In the workbench moderation upgrade context, this test is problematic because it works only in the case of a content that has no published revision. 
In case of a content saved as draft while a published revision exist, that tests the language against the published revision but not against the the revision that was just saved. 
So, it is not generic enough.

I recommend then to use use query that uses the "field_revision_{$field_name}" instead and use the current revision that goes with the node loading normally.

or rename the step to indicate that we test against the published revision or the draft revision when no published revision exists